lab_3/main.py

```diff
@@ -20,8 +20,31 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
-
+    if not isinstance(text, str) \
+            or not text:
+        return ()
+    split_regex = re.compile(r'[.|!|?|…]')
+    text = filter(lambda t: t, [t.strip() for t in split_regex.split(text)])
+    letters_result = []
+    shitty_symbols = ['`', '~', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '-', '+',
+                        '=', '{', '[', ']', '}', '|', '\\', ':', ';', '"', "'", '<', ',', '>',
+                        '.', '?', '/', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+    deutsch_buchstabe = {'ä': 'ae', 'ö': 'oe', 'ü': 'ue', 'ß': 'ss'}
+    for sentence in text:
+        sentence = sentence.lower()
+        for symbols in shitty_symbols:
+            sentence = sentence.replace(symbols, '')
+        tokens = sentence.split()
+        for word in tokens:
+            for letter, trans in deutsch_buchstabe.items():
+                word.replace(letter, trans)
+        letters = []
+        for token in tokens:
+            letters.append(tuple(['_'] + list(token) + ['_']))
+        if letters == []:
```

Comment:
Can be done without explicit comparison.

Reply:
Should it looks like that 
if not letters
Did I understand it right?